test/e2e-olm/kibana_test.go

```diff
@@ -0,0 +1,118 @@
+package e2e
+
+import (
```

Comment:
not sure what happened to my prior comment regarding this... but i envisioned we would move to a pattern more like what CLO does to verify components are running and aspects have been created[1][2].

This model is going to depend on previous tests being torn down correctly, or having enough resources for both tests to run in parallel (if that happens, i can't remember off the top of my head if thats been turned off). It worked when EO was only responsible for a single component, but it is going to be responsible for the ES parts, Kibana parts, curator parts (eventually), and the IndexManagement parts.

[1] https://github.com/openshift/cluster-logging-operator/blob/master/hack/testing-olm/test-010-deploy-via-olm-minimal.sh
[2] https://github.com/openshift/cluster-logging-operator/blob/master/hack/testing-olm/assertions

Reply:
Your previous comment is still [here](https://github.com/openshift/elasticsearch-operator/pull/300#discussion_r403341909). :point_up: 

Anyway I will continue the discussion here and resolve the thread above FYI to keep the focus here.

Regarding tear-down, AFAICS each create call through the E2E framework call registers a cleanup method. Once a testcase is down you need to call `ctx.Cleanup()` do tear down everything it was created. Parallel test execution, it is currently disabled according to [1].

Since we envision to move `oc.Create()/Exec()/...` calls and in general writing tests more in Golang, I would say let's not throw this bits of effort done here away. I full ack that setup/tear-down should be done per testcase and if not done properly it is a bug. Parallel tests are a separate discussion as it depends on resources we don't probably have on our CI. In extend the operator-sdk e2e test-framework we use under the hood provides wait functions for creation as well as for deletion and we can still wrap `oc.wait` where possible.

[1] https://github.com/openshift/elasticsearch-operator/blob/master/hack/testing-olm/test-001-operator-sdk-e2e.sh#L57